src/locale.ts

```diff
@@ -139,6 +125,65 @@ export function buildLocalisedChoice<T = string | number>(
 	return choice;
 }
 
+export function getInputLangStringOption(): SlashCommandStringOption {
+	return buildLocalisedCommand(
+		new SlashCommandStringOption().setRequired(false),
+		() => c("command-option").t`input-language`,
+		() => c("command-option-description").t`The language to search in, defaulting to the result language.`
+	).addChoices(...LOCALE_CHOICES);
+}
+
+export function getNameSubcommand(
+	getLocalisedDescription: () => string,
+	requiredStringOption?: SlashCommandStringOption
```

Comment:
Could theoretically extend to support an arbitrary number of extra options, but unlikely to have practical use case. Don't need premature abstraction.

Reply:
Yes, I could have made this an array. Because the builder doesn't have a generic addOption method though (only addStringOption, addUserOption, etc.), the generic solution is to accept a lambda that will be passed the builder as an argument.